Fix Bug 1180891, switch to new style newsletter subscription form fixes l10n problems

schalkneethling opened the pull request.

craigcook commented on media/css/plugincheck/plugincheck.less:
```diff
@@ -263,8 +263,100 @@ html[dir="rtl"] {
-    padding: 40px 20px;
+#mozorg-newsletter-form,
+#newsletter-form {
+    width: 1300px / 2;
```
This width is making it overflow in narrower windows. I think you don't need a width here at all and could just let it default to auto.
![plugincheck-overflow](https://cloud.githubusercontent.com/assets/205591/10763761/75bc51e0-7c88-11e5-9de5-0c65a01648f6.jpg)

craigcook commented on media/css/plugincheck/plugincheck.less:
```diff
@@ -263,8 +263,100 @@ html[dir="rtl"] {
+        width: auto;
+        padding: (@baseLine * 2) 0;
+        color: @textColorSecondary;
+        .font-size(46px);
```
This seems really big for an h3, even when centered as the main headline (see previous screenshot). I'd probably take this down to 32px.

craigcook commented on media/css/plugincheck/plugincheck.less:
```diff
@@ -263,8 +263,100 @@ html[dir="rtl"] {
+#newsletter-form {
+    width: 1300px / 2;
+    margin: 0 auto;
+    padding: 0 0 (@baseLine * 2) 0;
```
Could use some padding on the sides as well (see previous screenshot).

craigcook commented on media/css/plugincheck/plugincheck.less:
```diff
@@ -263,8 +263,100 @@ html[dir="rtl"] {
+        width: 50%;
+        margin: 0;
+
+        input[type="submit"] {
```
Should add `text-align: center;` for the button
![plugincheck-button](https://cloud.githubusercontent.com/assets/205591/10764221/488ea6da-7c8a-11e5-98d1-5e43ba62d80d.jpg)

schalkneethling commented:
thx @craigcook on it

schalkneethling commented:
@alexgibson r?

alexgibson commented:
Issue with the styling still?

![newsletter](https://cloud.githubusercontent.com/assets/400117/11066891/ed1f7200-87c1-11e5-85d2-4d9dc47777dc.png)

alexgibson commented on media/css/plugincheck/plugincheck.less:
```diff
@@ -263,8 +263,121 @@ html[dir="rtl"] {
+            background: #fbf9f4;
+            border: 2px solid #e1ddd3;
+            border-radius: 7px;
+            -webkit-appearance: none;
```
Does this also need `moz-appearance`, or is there a reason only webkit needs this?

schalkneethling replied:
Copied from elsewhere in bedrock but yeah, let me add that. There is no reason I am aware of not to.

alexgibson commented on media/css/plugincheck/plugincheck.less:
```diff
@@ -263,8 +263,121 @@ html[dir="rtl"] {
+        margin: 0;
+        padding: (@baseLine * 2) 0;
+        width: auto;
+        .font-size(2.4rem);
```
Nit: use `px` for consistency with the rest of the file

alexgibson commented on media/css/plugincheck/plugincheck.less:
```diff
@@ -263,8 +263,121 @@ html[dir="rtl"] {
+            text-align: center;
+            box-shadow: none;
+            border-radius: 7px;
+            -webkit-appearance: none;
```
Does this need `-moz-appearance` too?

alexgibson commented on media/css/plugincheck/plugincheck.less:
```diff
@@ -263,8 +263,121 @@ html[dir="rtl"] {
+}
+.newsletter-container {
+    background-color: #fff;
+    .border-radius(8px);
```
No need to use border radius mixin any more

schalkneethling replied:
Indeed. Should we also remove it from `lib.less` then? A little less code

schalkneethling replied:
Not in this bug of course, but in a separate bug/pr

alexgibson commented:
Thank you message needs responsive styles

![thankyou](https://cloud.githubusercontent.com/assets/400117/11067084/c3fc51a8-87c2-11e5-8d8b-c586f435805e.png)

schalkneethling commented:
@alexgibson ok, addressed all the issues mentioned here. The only thing is https://github.com/mozilla/bedrock/pull/3445#issuecomment-155458330

I could no replicate this with this form nor the non en-US one. Checked in Fx, Chrome as well as IE8 and 9

schalkneethling commented:
@alexgibson Ok, merged with master and fixed the form layout accordingly.

alexgibson commented:
Looks good, r+ - care to squash?
